suite-native/transactions/src/components/TransactionDetail/TransactionDetailSheets.tsx

```diff
@@ -1,84 +1,55 @@
 import React, { useState } from 'react';
-import { TouchableOpacity } from 'react-native';
 
-import { BottomSheet, Box, Card, Text, VStack } from '@suite-native/atoms';
-import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
-import { Icon, IconName } from '@trezor/icons';
+import { FiatCurrency } from 'suite-common/suite-config/src';
 
-const triggerStyle = prepareNativeStyle(utils => ({
-    flexDirection: 'row',
-    justifyContent: 'space-between',
-    paddingVertical: utils.spacings.small,
-}));
+import { Blockchain, WalletAccountTransaction } from '@suite-common/wallet-types/src';
+import { Card, VStack } from '@suite-native/atoms';
 
-const BottomSheetTrigger = ({
-    iconName,
-    title,
-    onPress,
-}: {
-    iconName: IconName;
-    title: string;
-    onPress: () => void;
-}) => {
-    const { applyStyle } = useNativeStyles();
-    return (
-        <TouchableOpacity style={applyStyle(triggerStyle)} onPress={onPress}>
-            <Box flexDirection="row">
-                <Box marginRight="medium">
-                    <Icon name={iconName} color="forest" />
-                </Box>
-                <Text>{title}</Text>
-            </Box>
-            <Icon name="chevronRight" />
-        </TouchableOpacity>
-    );
-};
+import { TransactionDetailParametersSheet } from './TransactionDetailParametersSheet';
+import { TransactionDetailValuesSheet } from './TransactionDetailValuesSheet';
+import { TransactionDetailInputsSheet } from './TransactionDetailInputsSheet';
 
 type SheetType = 'parameters' | 'values' | 'inputs';
 
-export const TransactionDetailSheets = () => {
-    const [expandedSheet, setExpandedSheet] = useState<SheetType | null>(null);
-
-    const handleCloseExpandedSheet = () => setExpandedSheet(null);
+type TransactionDetailSheetsProps = {
+    transaction: WalletAccountTransaction;
+    blockchain: Blockchain;
+    fiatCurrency: FiatCurrency;
+};
 
+export const TransactionDetailSheets = ({
+    transaction,
+    blockchain,
+    fiatCurrency,
+}: TransactionDetailSheetsProps) => {
+    const [expandedSheet, setExpandedSheet] = useState<SheetType | null>(null);
     return (
         <Card>
             <VStack spacing="small">
-                <BottomSheetTrigger
-                    iconName="warningCircle"
-                    title="Parameters"
-                    onPress={() => setExpandedSheet('parameters')}
+                <TransactionDetailParametersSheet
+                    isVisible={expandedSheet === 'parameters'}
+                    transaction={transaction}
+                    blockchain={blockchain}
+                    handleSheetVisibility={() =>
+                        setExpandedSheet(expandedSheet === 'parameters' ? null : 'parameters')
+                    }
```

Comment:
Create a handler that checks whether parameter sent to it is current expanded sheet and if not, call `setExpandedSheet(null)` to avoid pretty much same condition being written 3 times

Reply:
what about this?